addons/website/static/src/js/backend/button.js

```diff
@@ -111,6 +72,5 @@ var WidgetWebsiteButtonIcon = AbstractField.extend({
 });
 
 field_registry
-    .add('website_redirect_button', WidgetWebsiteButtonIcon)
-    .add('website_publish_button', WebsitePublishButton);
+    .add('website_redirect_button', WidgetWebsiteButtonIcon);
```

Comment:
Commit message: "This commit adapts the website_redirect_button and website_published_button to OWL."

Unless I totally misunderstand this whole commit, this does not yet convert `website_redirect_button` (but will if you do `website_redirect_button` and `website_redirect_button_legacy` as explained :+1:)

Reply:
The commit message was wrong however, there is already a `website_redirect_button` that was implemented here https://github.com/odoo/odoo/commit/9434969d5dc06aadcdbbfc600514a0bdb5635ea2 .

(Hence the tests already existing) I assume they kept the old one because of the view. 

For moving it to the module that still uses it, I'm not so certain which one still does. The way I found about the use of the legacy button was through a test. I removed the button and a test failed. That test was in the accounting module when editing a partner in a modal view.

I'm not confident enough to say that only these 2 modules use it, as a few views still need to be converted, and I'm not certain I know of all the flows to access the view that uses a `website_redirect_button`.

For the publish widget I feel confident that we can remove it as it's not as widely used.

I don't mind moving the legacy button in a separate `legacy` folder on website though if that's OK with you ?

Reply:
> I'm not confident enough to say that only these 2 modules use it, as a few views still need to be converted, and I'm not certain I know of all the flows to access the view that uses a website_redirect_button.

Well somebody will need to check all that at some point :wink: What's stopping you to check it ? :thinking: Is not this just about checking all uses in the code one by one (without having to know the flows) and if they are not in a custom view (js_class ?), then they can use the new one, otherwise they should use the old one? It would be useful to know as maybe the teams related to the old implementation may never care otherwise (because maybe they don't know).

Reply:
Just to clarify, the issue isn't that the button is used in a legacy or custom view. The issue is that the button is used in a view that can be rendered through a legacy widget (Usually the Form Dialog). 

In the case of res.partner, it's the res_partner_many2one widget that still hasn't been adapted. So any Studio form displaying a partner view could also be broken.

But the issue remain, I'm not sure how I'm supposed to find out which models can be called in that legacy FormView modal without checking every possible flow, or checking for each model that this button is on, if there's a field that reference them that is still rendered using the legacy view.

I'm sorry if this isn't completely clear, I have to admit that I haven't fully grasped how views are rendered yet.

If you have a suggestion on how I can find which module render this button in a legacy view, I wouldn't mind doing it though!

PS: I also asked the JS team on Discord if they had any advice on how to find which views are still rendered in legacy, and they told me that all views should be converted in 16.0 including studio if they're lucky. We could almost certainly remove this legacy button widget by then.

Reply:
So if a legacy widget instantiate a non-legacy view which contains a website_redirect_button, it will use the legacy field? That seems... strange :grimacing: But I don't know more than you on the subject.

But ok, if you are convinved you cannot possibly know for sure where legacy `website_redirect_button` are still needed, we can move on without it and we'll see later... but then the plan is to wait for res_partner_many2one to be converted then we remove the legacy field and... cross fingers? :grimacing:

Reply:
> So if a legacy widget instantiate a non-legacy view which contains a website_redirect_button, it will use the legacy field? That seems... strange 😬 But I don't know more than you on the subject.

The legacy widget will instantiate a legacy view. Since the Partner view isn't a custom view, it renders it using the legacy FormView widget, and rendering the legacy buttons with it.

You can actually create the same issue with the website_publish_button that I removed. Its defined in the `gamification_badge_view_form` template. If I go in studio and make a many2one field that uses the res_partner_many2one widget  and link it to Gamification badge, the widget will not be loaded properly.
I just think it's so niche that it's OK to remove the legacy widget. Especially if the OWL team's goal is to converted all these widgets in 16.0.

Indeed, we'll have to cross fingers 🤞🏻  when we remove the button, or somehow find a way to make sure that we can no longer invoke those views via the legacy widget system.